pkg/executor/sortexec/sort_spill.go

```diff
@@ -114,12 +114,13 @@ func (s *parallelSortSpillAction) actionImpl(t *memory.Tracker) bool {
 		s.spillHelper.cond.Wait()
 	}
 
+	// we may be in `needSpill` status, so it's necessary check if we are in `notSpilled` status
```

Comment:
unrelated changes?

Reply:
> unrelated changes?

Yes, it's unrelated with hash join refine. But I find that we can refine the comment in sort spill, so I put it in this pr.